db/pool.go

```diff
@@ -0,0 +1,200 @@
+package db
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"database/sql"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/jackc/pgx"
+	"github.com/jackc/pgx/stdlib"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/sean-/vpc/internal/buildtime"
+	"github.com/sean-/vpc/internal/logger"
+)
+
+type Pool struct {
+	pool            *pgx.ConnPool
+	stdDriverConfig *stdlib.DriverConfig
+	config          Config
+}
+
+func New(cfg Config) (*Pool, error) {
+	pool := &Pool{
+		config: cfg,
+	}
+
+	tlsConfig, err := cfg.TLSConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to generate a TLS config")
+	}
+
+	// TODO(jen20) do we even want to support password auth vs TLS?
+	poolConfig := pgx.ConnPoolConfig{
+		ConnConfig: pgx.ConnConfig{
+			Logger:   logger.NewPGX(log.Logger),
+			Database: cfg.Database,
+			User:     cfg.User,
+			//Password: cfg.Password,
+			Host: cfg.Host,
+			Port: cfg.Port,
+			Dial: (&net.Dialer{Timeout: cfg.ConnTimeout, KeepAlive: 5 * time.Minute}).Dial,
```

Comment:
The 5min keepalive should be moved to a constant since I think that value is being reused somewhere.